libs/cardano-ledger-core/src/Cardano/Ledger/BaseTypes.hs

```diff
@@ -844,3 +846,21 @@ class Inject t s where
 -- | Helper function for a common pattern of creating objects
 kindObject :: Text -> [Pair] -> Value
 kindObject name obj = object $ ("kind" .= name) : obj
+
+-- | Loseless way of representing BoundedRational as JSON that uses decimals when feasible
+toRationalJSON :: BoundedRational a => a -> Value
+toRationalJSON br =
+  case fromRationalRepetendLimited 20 r of
+    Right (s, Nothing) -> toJSON s
+    _ -> toJSON r
+  where
+    r = unboundRational br
+
+-- | Aeson parser that works with the result of toRationalJSON
+-- | (It adapts a BoundedRational so that it uses the Rational parser instead)
+parseAsRational :: BoundedRational r => Parser Rational -> Parser r
+parseAsRational parser = do
+  rational <- parser
+  case boundRational rational of
+    Nothing -> fail "Cannot convert to BoundRational"
+    Just x -> return x
```

Comment:
This function isn't needed. `FromJSON` instance for all bounded types will already do that.